Program.cs

```diff
@@ -1,20 +1,96 @@
 ﻿// ---------------------------------------------------------------------------------------
 // Training ~ A training program for new joinees at Metamation, Batch- July 2023.
 // Copyright (c) Metamation India.                                                
-// ---------------------------------------------------------------------------
+// ---------------------------------------------------------------------------------------
 // Program.cs                                                                     
-// 
-// 
-// 
+// Program to create a custom MyList<T> class.
+// Class is created with arrays as the underlying data structure.
+// Must start with an initial capacity of 4 and double its capacity when needed.
+// Throw exceptions when necessary. 
 // ---------------------------------------------------------------------------------------
-using static System.Console;
+
 namespace Training {
-   #region class Program ------------------------------------------------------------------------
+   #region class Program --------------------------------------------------------------------------
    internal class Program {
-      #region Method ----------------------------------------------
-      static void Main (string[] args) {
-         Console.WriteLine ("Hello, World!");
+      #region Method ------------------------------------------------
+      static void Main () { }
+      #endregion
+   }
+
+   /// <summary>Custom MyList<T> class</summary>
+   public class MyList<T> {
+      #region Properity ---------------------------------------------
+      /// <summary>Gets the count of the MyList</summary>
+      public int Count => mCount;
+
+      /// <summary>Gets the size of the MyList</summary>
+      public int Capacity => mList.Length;
+
+      /// <summary>Gets the element or sets the value in the given index of MyList</summary>
+      /// <param name="index">Any index of MyList</param>
+      public T this[int index] {
+         get => mList[index];
+         set => mList[index] = value;

```

Comment:
Exception should be thrown when the user tries to access the index that is out of range.